src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,170 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Represents the component of the Duke program
+ * that makes sense of user input.
+ */
+public class Parser {
+
+    protected static DateTimeFormatter INPUT_DATETIME_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+
+    /**
+     * Returns a valid Duke Command after parsing user input.
+     * @param input user input.
+     * @return a valid Duke Command
+     * @throws DukeException when user input
+     *         does not match any valid Duke Commands.
+     */
+    public Command parseCommand(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        try {
+            return Command.valueOf(arr[0].toUpperCase());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new DukeException("OOPS!!! I'm sorry, but that is an invalid command :-(");
+        }
+    }
+
+    /**
+     * Returns a ToDo based on user input.
+     * @param input user input.
+     * @return a ToDo based on user input.
+     * @throws DukeException when user input does not contain
+     *         the description of the ToDo after the command.
+     */
+    public ToDo parseToDo(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        if (arr.length < 2) {
+            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+        }
+        return new ToDo(input.substring(5));
+    }
+
+    /**
+     * Returns a Deadline based on user input.
+     * @param input user input.
+     * @return a Deadline based on user input.
+     * @throws DukeException when user input does not contain
+     *         the description of the Deadline after the command,
+     *         the date and time of the Deadline after '/by'
+     *         or when the date and time is not in the correct format.
+     */
+    public Deadline parseDeadline(String input) throws DukeException {
+        String[] arr = input.split(" ");
+        String[] arr2 = input.split(" /by ");
```

Comment:
Hi, in terms of code quality, perhaps we can consider to use words that are more descriptive for these variables instead of arr and arr2.

Reply:
I agree with enhao25! Variable names are important to convey to the reader what your code is actually doing and what the variable is for. Perhaps, consider renaming them to names that fit the context.